concepts/arrays/about.md

```diff
@@ -0,0 +1,143 @@
+# About
+
+JSON defines an array as:
+
+> An array is an ordered collection of values.
+> An array begins with `[` left bracket and ends with `]` right bracket.
+> Values are separated by `,` comma.
+
+Arrays can contain zero or more of _any kind_ of JSON value.
+
+## Creating arrays
+
+Use brackets to collect elements into an array
+```jq
+[1, 2, 3]
+```
+
+Inside the brackets can be an expression that generates a _stream_ of elements
+```jq
+[range(5)]   # => [0, 1, 2, 3, 4]
+```
+
+## Size
+
+The `length` function returns the number of elements in the array.
+
+## Indexing and slicing
+
+Array indexing is zero-based.
+
+Retrieve an element from an array with a bracket expression:
+`.[2]` gets the third element.
+
+Negative indexes count backwards from the end of the array:
+`.[-1]` gets the last element; `.[-2]` is the second last.
+
+A "slice" is a sub-sequence of the array.
+.`[10:15]` returns 5 elements starting from index 10; the end index is _not included_.
+
+There are some convenience functions:
+
+- `first` gets the first element,
+- `last` gets the last element,
+- `nth(n)` gets the element at index `n`
+
+## Convert to a stream of elements
+
+There are circumstances where we want to stream the elements of an array individually.
+For this we use `.[]`.
+
+Suppose we want a stream of the lengths of the strings in an array
+
+```jq
+["a", "be", "cat", "door"] | length        # => 4, we don't want the length of the array
+["a", "be", "cat", "door"] | .[] | length  # => the stream 1, 2, 3, 4 -- the lengths of each element
+["a", "be", "cat", "door"][] | length      # => same as above
+```
+
+## Concatenating and Subtracting
+
+Use `+` to concatenate arrays
+
+```jq
+[1, 2, 3] + [4, 5]     # => [1, 2, 3, 4, 5]
+```
+
+The `-` operator removes elements in the right-hand array from the left-hand array.
+
+```jq
+[range(10)] - [2, 4, 6]     # => [0, 1, 3, 5, 7, 8, 9]
+```
+
+## Enumeration
```

Comment:
Iteration?

In Python, at least, enumerate means listing values ... alongside their index.

Reply:
I hear you. I was using [the Ruby sense](https://ruby-doc.org/3.1.3/Enumerable.html)